packages/lisk-transaction-pool/src/job.ts

```diff
@@ -0,0 +1,50 @@
+export class Job<T> {
+	private _id: NodeJS.Timer | undefined;
+	private readonly _interval: number;
+	private readonly _job: () => Promise<T>;
+	private _running = false;
+
+	public constructor(
+		job: () => Promise<T>,
+		interval: number,
+	) {
+		this._interval = interval;
+		this._job = job;
+	}
+
+	public async start(): Promise<void> {
+		if (!this._running) {
+			this._running = true;
+
+			return this.run();
+		}
+	}
+
+	public stop(): void {
+		if (this._running && this._id  !== undefined) {
+			clearTimeout(this._id);
+			this._id = undefined;
+			this._running = false;
+		}
+	}
+
+	private async callJobAfterTimeout(): Promise<void> {
+		return new Promise<void>(resolve => {
+			this._id = setTimeout(async () => {
+				await this._job();
+
+				return resolve();
+			}, this._interval);
+		});
+	}
+
+	private async run(): Promise<void> {
+		// Base case for recursive function
+		if (!this._running) {
```

Comment:
Not the other way around? This condition means I can never run a job if it's not running.

Reply:
Once you call `start` at least once, the run function will be checking `if (!this._running) ` condition indefinitely. Calling `stop` doesn't revert the Job to the initial state. You could achieve it by getting rid of recursion in favour of while:

```suggestion
		while (this._running) { // I would suggest _started or _active
		    await this.callJobAfterTimeout();
		} 
```
Lmk what do you think.